google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -1962,6 +2276,31 @@ public List<ByteArray> getBytesArray() {
       }
     }
 
+    @Override
+    public <T extends AbstractMessage> List<T> getProtoMessageArray(T m) {
+      Preconditions.checkNotNull(
+          m,
+          "Proto message may not be null. Use MyProtoClass.getDefaultInstance() as a parameter value.");
+      checkNotNull();
+      try {
+        List<T> protoMessagesList = new ArrayList<>(value.size());
+        for (LazyByteArray protoMessageBytes : value) {
+          if (protoMessageBytes == null) {
+            protoMessagesList.add(null);
+          } else {
+            protoMessagesList.add(
+                (T)
+                    m.toBuilder()
+                        .mergeFrom(protoMessageBytes.getByteArray().toByteArray())
```

Comment:
Here also: Would it be possible to apply the same kind of optimization as above?

Reply:
I have added the optimization suggested above to this.